src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -239,20 +225,48 @@ create_distributed_table(PG_FUNCTION_ARGS)
 
 	relation_close(relation, NoLock);
 
-	char *distributionColumnName = text_to_cstring(distributionColumnText);
-	Assert(distributionColumnName != NULL);
+	bool shardCountIsStrict = false;
+	if (distributionColumnText)
```

Comment:
I don't understand why we have this condition. Maybe leave that out of this PR until we actually introduce logic related to distributionColumnText == NULL.

Reply:
moved these to https://github.com/citusdata/citus/pull/6745